package/spec/frontend/browser/agent_spec.js

```diff
@@ -73,7 +73,123 @@ describe('pageflow.browser.Agent', function() {
         'AppleWebKit/602.1.50 (KHTML, like Gecko) Version/10.0 Mobile/14A403 Safari/602.1'
       );
 
-      expect(agent.matchesDesktopSafari()).toBe(false);
+      expect(agent.matchesDesktopSafari({minVersion: 3})).toBe(false);
+    });
```

Comment:
Can you add tests for the case when `minVersion` option is not present? I'm not sure, but I think `matchesDesktopSafari` now always returns false if the version option is not present.

Reply:
It does return false. Added tests for that.

Reply:
But that's wrong, isn't it? It *is* a desktop Safari, we just did not give a min version.

Reply:
[Existing code like this](https://github.com/codevise/pageflow/blob/f382c40084e3bff3413ceef0e2864179af57b0b6/package/src/frontend/browser/volumeControlSupport.js#L9) depends on `agent.matchesDesktopSafari()` returning `true` on a Desktop Safari. With your change, this call results in an error now. And even when passing `agent.matchesDesktopSafari({})` it now returns `false`. The test suite no longer shows that since you also changed the tests that have nothing to do with min version to pass that option as well.

I'd remove the option from all other tests and move the tests from the `describe('#minVersion', ...)` block into the describe blocks for the individual methods. The `#` normally means that the following string is the name of a method on the tested class (e.g. `Agent#matchesDesktopEdge`) - but there is no `Agent#minVersion` method.  If you look at those spec descriptions as a sentence, you see that they do not capture why the return value is the way it is:

"pageflow.browser.Agent#minVersion returns true for Chrome 83"

Instead I'd go for something like

"pageflow.browser.Agent#matchesDesktopFirefox returns false for Firefox 17 when minVersion 20 is given"

Reply:
You mean to change `matchesMinVersion` in `Agent` , like to add a condition in `matchesMinVersion` , that if `minVersion` option is not given then there should be a default min version value and then `machesDesktopSafari` will always return true?

Reply:
Yes, I should have considered the fact that it will affect the existing code where theses functions are being used.
Now, I took care of the possibility that if no `minVersion` is given that `agent.mathcesDesktopSafari` should return true.

Removed the option from existing test cases and have moved `minVersion` tests in their respective methods.